### ISSUE TITLE
Should fading work?

### ISSUE THREAD

**Issue by maelle** [rstudio/bslib#324]
In a [PR to pkgdown](https://github.com/r-lib/pkgdown/pull/1694) where we now use bslib for BS4 dependencies, I'm adding the HTML structure for tabsets including support for fading as documented in an R Markdown book.

I'm following the [Bootstrap docs](https://getbootstrap.com/docs/4.6/components/navs/#fade-effect).

Here's the HTML I've created: https://60b8c9c243e3b70d06c196c1--pkgdown-dev.netlify.app/dev/articles/test/rendering.html#results-in-tabset-fading-1 where the second tab-panel has the active and show classes, the other panel has the fade class.

```html
<div id="results-in-tabset-fading" class="section level2 tabset tabset-fade">
<h2 class="hasAnchor">
<a href="#results-in-tabset-fading" class="anchor" aria-hidden="true"></a>Results in tabset, fading</h2>


<ul class="nav nav-tabs nav-row" id="results-in-tabset-fading">
<li class=""><a data-toggle="tab" href="#something" role="tab">Something</a></li>
<li class="active"><a data-toggle="tab" href="#nice" role="tab">Nice</a></li>
</ul>
<div class="tab-content">
<div id="something" class="fade tab-pane" role="tabpanel">

<p>ok</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="rendering_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<div id="cool-1" class="section level4">
<h4 class="hasAnchor">
<a href="#cool-1" class="anchor" aria-hidden="true"></a>cool</h4>
<p>yay</p>
</div>
</div>
<div id="nice" class="show active fade tab-pane" role="tabpanel">

<p>bonjour</p>
</div>
</div>
</div>
```

Now if I click on the panels it removes their "fade" class so I see no fading.
I've tried removing the bslib dependencies in favor of CSS+JS for Bootstrap via CDN and I see the fading https://codepen.io/maellesalmon/pen/rNydawa (I've made it super obvious in the pen with a very slow transition).
I guess I am yet again missing something in bslib, and would appreciate any hint. Thanks for the great package in any case!

**Comment by cderv** [comment]
It seems there is something indeed with BS4. This does not work as it should I think in rmarkdown

````markdown
---
title: "RMarkdown tabset fade"
output: 
  html_document:
    self_contained: false
    theme:
      version: 4
---

## Quarterly Results {.tabset .tabset-fade}

### By Product

(tab content 1)

### By Region

(tab content 2)
````

Try with
```
    theme:
      version: 3
```
and without bslib , you'll see the fading.

With BS4, the two divs for the tab content loose their class `.fade`

````html
<div class="tab-content">
<div id="by-product" class="section level3 tab-pane tabbed-pane active" role="tabpanel">
<p>(tab content 1)</p>
</div>
<div id="by-region" class="section level3 tab-pane tabbed-pane" role="tabpanel">
<p>(tab content 2)</p>
</div>
</div>
````

If I add it manually in the browser, it disappears on the next click on a tab. 

Something in the bs3compat layer ?

**Comment by cpsievert** [comment]
This is a duplicate of #6 (currently, you'll have to use 'valid' BS4 markup to get fading to work). Can pkgdown emit BS4 markup or does it need to be compatible with BS3? If so, I'll try to fix this in bslib (otherwise, for rmarkdown, it might be easier to just fix this issue in rmarkdown itself)

**Comment by maelle** [comment]
I thought the pkgdown markup was BS4, I'll have to re-check then!

**Comment by cpsievert** [comment]
The markup in your comment is BS3, which is what you want to emit if you need to support Bootstrap 3 and up (we're pretty committed to supporting it in Shiny and R Markdown, which is why `bs_theme(version = 4)` includes an extra 'compatibility' layer to recognize both forms).

**Comment by cpsievert** [comment]
By the way, I'm currently working on Bootstrap 5 support (#82) and this appears to no longer be an issue, so I might just let this be a known Bootstrap 4 issue.

(Once Bootstrap 5 support lands `bs_theme()` will start defaulting to `bs_theme(version = 5)` since I'd like to move away from Bootstrap 4 as fast as possible)

**Comment by maelle** [comment]
@cpsievert Thank you! How do you recognize it's BS3? 

Thanks for the information reg Bootstrap 5. Do you mean you won't support BS4 any more?

**Comment by maelle** [comment]
(I'll fix my markup with https://getbootstrap.com/docs/4.6/components/navs/#javascript-behavior :innocent: :see_no_evil: )

**Comment by cderv** [comment]
Good to know that R Markdown tabset fading won't work if you use BS4. I did not have that in mind. Thanks for asking @maelle !

**Comment by cpsievert** [comment]
> Thank you! How do you recognize it's BS3?

The active class is on the `<li>` and there is no `.nav-item` or `.nav-link`

> Do you mean you won't support BS4 any more?

I think we'll consider `version = 4` superceded by `version = 5`, meaning that you can continue using `version = 4`, but we won't be actively developing it (only making critical bug fixes)

**Comment by maelle** [comment]
Thank you for the information! :pray:

**Comment by cpsievert** [comment]
After looking into this more, I think it's going to make sense to fix this across all Bootstrap versions, see https://github.com/rstudio/bslib/pull/325 for a tentative fix

**Comment by cderv** [comment]
That works now in Rmd. Thanks!
